packages/nimble-components/src/wafer-map/specs/features/change-highlight.md

```diff
@@ -0,0 +1,230 @@
+
+![retrieved-metrics](resources/retrieved-metrics.PNG)
+
+You can see the IDs and the corresponding values of Wafer (B9IQ8...), DieX (90) and DieY (82)
+
+The component can highlight only based on the value that appears on the die. For example here:
+
+![dies-values](resources/dies-values.PNG)
+
+We wish to not be constrained by the values appearing on the die but to be able to highlight based on what we receive from the selection function
+
+## Links To Relevant Work Items and Reference Material
+
+[MicroStrategy Enable Custom Selection and Highlighting](https://www2.microstrategy.com/producthelp/Current/VisSDK/Content/topics/HTML5/selection_highlightAPI.htm)
+
+[Introduction to Attributes, Metrics, and Groups](https://www2.microstrategy.com/producthelp/Current/MSTRWeb/WebHelp/Lang_1033/Content/CreatingNewObjects.htm)
+
+## Implementation / Design
+
+We will replace the old logic, removing it completely, and add a new one.
+
+Each die will have an optional field called "tags", being optional in the cases when we don't use highlighting we can get rid of it.
+
+```ts
+export interface WaferMapDie{
+...
+   tags?: string[];
+...
+}
+```
+
+How the tags looks like remain a choice of the user, they can either have a simplified form
+
+```ts
+['a', 'b', 'c'];
+```
+
+Or can be the id of the attributes
+
+```ts
+[
+    'h90;85C4BC04377947939B9B8E62A7AC9BA5',
+    'h530;208BD62A39A84D96825CEBF10B1E6179',
+    'h80;A1DEB28673CA4A2B8C8DE26461D2FFB7'
+];
+```
+
+We will make the highlights based on this tags. The wafer will receive a list of highlightedTags
+
+```ts
+@observable public highlightedTags: string[]
+```
+
+Example of flow :
+
+```ts
+export const dies: WaferMapDie[] = [
+    {
+        x: 0,
+        y: 0,
+        value: '14',
+        tags: ['a']
+    },
+    {
+        x: 1,
+        y: 0,
+        value: '76',
+        tags: ['a', 'b']
+    },
+    {
+        x: 0,
+        y: 1,
+        value: '72'
+    }
+];
+```
+
+A die will be highlighted if any of the die's tags equals at least one value in the highlightedTags
+
+If the highlightedTags equals
+
+```ts
+['a'];
+```
+
+dies[0][0] will be highlighted because it contains 'a'
+
+dies[1][0] will be highlighted because it also contains 'a'
+
+dies[0][1] will not be be highlighted because it does not have any tag
+
+```ts
+['b', 'h'];
+```
+
+dies[0][0] will not be highlighted because it does not contain 'b' neither 'h'
+
+dies[1][0] will be highlighted because it contains 'b'
+
+dies[0][1] will not be be highlighted because it does not have any tag
+
+If the highlightedTags equals
+
+```ts
+[];
+```
+
+Nothing will be highlighted
+
+If the highlightedTags equals
+
+```ts
+['efg'];
+```
+
+All dies will be faded
+
+Note: Besides the code changes we will also need to update the tests and nimble story book.
```

Comment:
Is the tag empty string a valid tag for a wafer to define?
```
{
        x: 0,
        y: 0,
        value: '14',
        tags: ['']
    },
```

Is the tag empty string a valid tag for highlightedTags to define?
```
['']
```

I think it would be fine to do so. No reason to add complexity to block it. We should just have tests for that case. Can you make a section of expected test cases to cover?

Reply:
Once you and @munteannatan agree the expected test cases section covers the edge cases needed feel free to merge. We'll reference those cases when reviewing tests in future submissions.